src/Types.hs

```diff
@@ -34,27 +34,27 @@ import qualified Prelude
 
 data ScriptParams = ScriptParams
         {
-              pNftCs :: CurrencySymbol
+              pNftCs :: CurrencySymbol -- ^ CurrencySymbol of the ENNFT
         } deriving (Prelude.Show, Generic, FromJSON, ToJSON, Prelude.Eq, Prelude.Ord)
 PlutusTx.unstableMakeIsData ''ScriptParams
 PlutusTx.makeLift ''ScriptParams
 
 data EnRegistration = EnRegistration
         {
-              enOperatorAddress :: BuiltinByteString
-            , enConsensusPubKey :: BuiltinByteString
-            , enMerkleTreeRoot  :: BuiltinByteString
-            , enCceAddress      :: BuiltinByteString
-            , enUsedNftTn       :: TokenName
-            , enRwdWallet       :: PubKeyHash
-            , enCommission      :: Integer
+              enOperatorAddress :: BuiltinByteString -- ^ TODO Address of what exactly? Cardano address?
+            , enConsensusPubKey :: BuiltinByteString -- ^ TODO What is that?
+            , enMerkleTreeRoot  :: BuiltinByteString -- ^ TODO What is this used for?
+            , enCceAddress      :: BuiltinByteString -- ^ TODO What is CCE?
+            , enUsedNftTn       :: TokenName -- ^ TODO Why "used"? This is the ENNFT right? And IIUC, the token name is specific to the operator?
+            , enRwdWallet       :: PubKeyHash -- ^ TODO So that is the wallet the operator should get the rewards after participating in block production in Aya?
+            , enCommission      :: Integer -- ^ TODO What is that exactly?
```

Comment:
Most of this parameter are going to change as they were cosmos implementation specific.

`enOperatorAddress` -> Owner Account which is operating the Aya Validator on the Aya chain
`enConsensusPubKey` -> Validator Pubkey of the consensus Node
`enMerkleTreeRoot` -> MerkleRoot for verifiable randomness (this will disappear)
`enCceAddress` -> CrossChain Pubkey which can be verified on both chains
`enUsedNftTn` -> Unique ENNFT name, "used" for this registration, what does IIUC mean?
`enRwdWallet` -> yes 
`enCommission` -> commission in percent shared with staking delegators.